docs/decisions/2023-06-05_github_merge_queue.md

```diff
@@ -0,0 +1,49 @@
+# 2023-06-05 GitHub Merge Queue
+
+* status: proposed
+* date: 2023-06-05
+* authors: Brian Samek
+
+## Context and Problem Statement
+
+* [Epic](https://go/github-merge-queue)
+* [Scope](https://go/github-merge-queue-scope)
+* [Design](https://go/github-merge-queue-design)
+
+GitHub merge queue integrates natively with GitHub Actions. Integration with
+other CI systems exists but is lightly documented. The
+[docs](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/managing-a-merge-queue#triggering-merge-group-checks-with-other-ci-providers)
+state that a CI system should listen for branches matching a pattern and that
+the PR will be merged once required checks pass.
+
+The purpose of this ADR is to keep a record of the details of the integration
+between the GitHub merge queue and Evergreen for later reference.
+
+## Architectural Overview
+
+Evergreen listens for new GitHub merge queue items, creates and runs versions
+for these items, and posts the results of these versions to GitHub.
+
+### Listen
+
+Evergreen listens for new branches matching `gh-readonly-queue/{base_branch}`.
+
+### Create version
+
+Evergreen creates a new version with a new requester type "github-merge-queue".
+There are two alternative possibilities: First, use the ad hoc version type.
+However, the ad hoc version type cannot independently be given increased
+priority in the scheduler. Using ad hoc overloads the meaning of this type.
+Second, use the commit queue type. However, this type triggers a substantial
+amount of Evergreen commit queue logic, so it's more straightforward to make a
+new type. A disadvantage is that anyone downstream who has written queries that
+assumes the number of requesters is fixed will need to take into account the new
+requester.
+
+### Post results
+
+Post a check called "evergrene/github-merge-queue.  Although "evergreen" might
```

Comment:
Is this a check in an official GitHub checks sense or is it just a GitHub PR status like what's usually posted? The distinction is [here](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/collaborating-on-repositories-with-code-quality-features/about-status-checks#types-of-status-checks-on-github).

Reply:
I'm not sure. I can update the ADR later when I know more. (I intend to update it as I make other PRs.) The GitHub support person I asked originally about this explicitly said the checks API, but the docs say "status checks", and from what I know about branch protection rules these are statuses.

Reply:
After some discussion with Chaya, I believe that it's in the checks sense. I updated the text.